files/default/sql/ddl/2.1.0__initial_tables.sql

```diff
@@ -1014,21 +1014,39 @@ CREATE TABLE `project` (
   `created` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
   `retention_period` date DEFAULT NULL,
   `archived` tinyint(1) DEFAULT '0',
-  `conda` tinyint(1) DEFAULT '0',
   `logs` tinyint(1) DEFAULT '0',
   `deleted` tinyint(1) DEFAULT '0',
-  `python_version` varchar(25) COLLATE latin1_general_cs DEFAULT NULL,
   `description` varchar(2000) COLLATE latin1_general_cs DEFAULT NULL,
   `payment_type` varchar(255) COLLATE latin1_general_cs NOT NULL DEFAULT 'PREPAID',
   `last_quota_update` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
   `kafka_max_num_topics` int(11) NOT NULL DEFAULT '100',
   `docker_image` varchar(255) COLLATE latin1_general_cs DEFAULT NULL,
+  `python_env_id` int(11) DEFAULT NULL,
   PRIMARY KEY (`id`),
   UNIQUE KEY `projectname` (`projectname`),
   UNIQUE KEY `inode_pid` (`inode_pid`,`inode_name`,`partition_id`),
   KEY `user_idx` (`username`),
   CONSTRAINT `FK_262_290` FOREIGN KEY (`username`) REFERENCES `users` (`email`) ON DELETE NO ACTION ON UPDATE NO ACTION,
-  CONSTRAINT `FK_149_289` FOREIGN KEY (`inode_pid`,`inode_name`,`partition_id`) REFERENCES `hops`.`hdfs_inodes` (`parent_id`,`name`,`partition_id`) ON DELETE CASCADE ON UPDATE NO ACTION
+  CONSTRAINT `FK_149_289` FOREIGN KEY (`inode_pid`,`inode_name`,`partition_id`) REFERENCES `hops`.`hdfs_inodes` (`parent_id`,`name`,`partition_id`) ON DELETE CASCADE ON UPDATE NO ACTION,
+  CONSTRAINT `FK_362_309` FOREIGN KEY (`python_env_id`) REFERENCES `python_environment` (`id`) ON DELETE SET NULL ON UPDATE NO ACTION
```

Comment:
Since we now have a new table for keeping track for python envs for projects, having a ` UNIQUE KEY `project_env` (`project_id`,`python_version`),` means we can have multiple environments per project. How does this FK help in maintaining constraint integrity? It seems it forces a project to have one environment or am I missing something? The child table is `python_environment`, it is an index scan to find the python environment of a project which is good enough performance wise I believe

Reply:
I changed the UNIQUE KEY to only be used on the project_id now. As we don't have the backend implementation for multiple environments I don't see a direct need to model the database for that right now. The old key was a leftover.

Reply:
Still, when will the `project.python_env_id` be used? 
The project's env is derived from the dimension table `python_environment` and when a project is removed, the environment will be removed due to the FK. 

In JPA there owning side is PythonEnviromnent and in Project there will be the mapped entity (PythonEnvironment).

Similar to 
```
@OneToMany(cascade = CascadeType.ALL,
      mappedBy = "projectId")
  private Collection<JupyterProject> jupyterProjectCollection;
```
but with OneToOne.

Reply:
You're right there is not a need to have a foreign key in this case. I got rid of it.